### DISCUSSION TITLE
How to disable new device authorization?

### DISCUSSION THREAD

**Discussion by ghost** [community/community#41905] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Question

### Body

GitHub sends code to my email.

It is annoying and does not make any benefits for me personally.

**Comment by jfullstackdev** [comment]
you mean you have so many devices that you are logging in ? then you start being annoyed ? 

usually that will only be your phone and your laptop, I guess

**Reply by ghost** [reply]
It is not only annoying, for some reason email can be temporarily unavailable.

But yes, I have a lot of devices, I have a lot of browsers.
And I do not stay logged in to my accounts. I always use empty context to login each time when I need that.

Though all that does not matter because "no alternative/no option" is bad way.

**Comment by airtower-luna** [comment]
The alternative is to enable two factor authentication (2FA), see: https://docs.github.com/en/authentication/securing-your-account-with-two-factor-authentication-2fa/configuring-two-factor-authentication

If you use TOTP or a security key you don't need access to any other online service.

**Reply by ghost** [reply]
I am trying to avoid 2FA at all including for new device and you suggest to use 2FA all the time 😁

**Reply by airtower-luna** [reply]
It's the only other option you have, and you won't need email access that way. :sweat_smile: 

Mind that some time next year [2FA will become mandatory](https://github.blog/2022-05-04-software-security-starts-with-the-developer-securing-developer-accounts-with-2fa/) for anyone who contributes code anyway.

**Reply by ghost** [reply]
Thank you for information, I afraid that GitHub will repeat Skype way, and it happens now. Good time to consider about alternatives.
